### DISCUSSION TITLE
Send live data to Web-Api

### DISCUSSION THREAD

**Discussion by MaximilianSteverding** [tbnobody/OpenDTU#1122] [Ideas]
I would like to post the live data (updated live data of the Inverter) to a Web-Server to further process the data there. It would be great if there would be an option to type a web-address to which the OpenDTU posts a JSON object or a get message with its location, current output and timestamp. Could something like that be easily integrated or could you hint me where I would be able to integrate it myself? Looking forward to your answers!

**Comment by ituri** [comment]
You mean the [web API](https://github.com/tbnobody/OpenDTU/blob/master/docs/Web-API.md)?

**Reply by MaximilianSteverding** [reply]
As far as I understand the Web API is for an external request to the OpenDTU - however my OpenDTU is not exposed to the web and I cannot access it from a remote server outside of my network. Therefore I would much rather like to have the option to type in a Webadress to which the DTU posts the values on a regular basis, that can also be outside of my network anywhere on the web.

**Reply by ituri** [reply]
I don't really feel that this is something that should be implemented with some code on the ESP32. After all, this is a networking issue. 
You could try exposing just the API endpoint and securing it with HTTP auth. I've done so myself in order to pull information for a widget on my phone.

Maybe also try the MQTT route and then look at projects like [mqtt-http-bridge](https://github.com/njh/mqtt-http-bridge).

**Comment by Tobi266** [comment]
I think MaximilianSteverding is searching something like a http push function.
I use this on some Tasmota devices and it works great.
And the biggest advantage, I don't have to run an MQTT broker.
The ESP simply sends a value to an HTTP directory in JSON format when the power values change.
So I don't have to spend extra CPU Power on another device...

**Reply by MaximilianSteverding** [reply]
That is exactly what I was looking for / would want to implement, because you do not need the overhead of a broker etc. but only a webadresss that you can push the data to which can then process the data. Could be as simple as sending all the values the DTU receives from the inverter in JSON -format to a webadress that one could enter in a separate field. The server that you would send the data to could than be programmed in anyway you would want to process the data. Main reason is that I do not want to setup another device in my home network only to broker the information nor I want to expose / can expose (due to lacking static IP Adress) my network so that I can access the DTU externally. Anyway such a feature should be easygoing integratable right?

**Comment by oldieman** [comment]
There is another way to get the data from the DTU outside your home network. Just get a NOIP.COM account redirect it to your Modem. In the modem open a port and redirect this port form outside to your address of the DTU with port 80.
Than you can get your data with the following HTTP Get call:
http://xxxx/api/livedata/status
where xxxx is the URL of your domain on NOIP or locally with your IP-Address of your DTU. The data are in JSON format.
I hope it helps you.

**Comment by simsasaile** [comment]
I'll jump in here, because I would also like to see the functionality to send data to a web server. Specifically, I'm talking about an emonCMS server. As also mentioned here, this is super flexible with Tasmota scripts. I use this to send the values from the smart meter interface directly to the emoncms instance, how this is configured I have described here: https://community.openenergymonitor.org/t/read-smart-meters-sml-interface-with-esp-and-tasmota-and-send-data-to-emoncms/23642  If now also the PV generation can be sent there with OpenDTU (without the need for yet another device with MQTT broker) this opens a inexpensive self-consumption monitoring option for a lot of people to whom I recommended a balcony power plant, but who are not technophile.

Therefore I would be very happy to have an option to send data directly to a web server.

**Comment by FrozenAssassine** [comment]
Any updates on this? Would also like to have this feature implemented for a much simpler upload of the api data. Currently I have another Pc uploading the data to my server, because port opening is not working for me.
